In [1]:
import pandas as pd
from datetime import datetime
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
# 시각화 과정에서 한글 깨짐 방지!!!!

from matplotlib import font_manager, rc
import matplotlib
import platform
import matplotlib.pyplot as plt

%matplotlib inline 

if platform.system() == 'Windows':
    font_name = font_manager.FontProperties(fname="c:/Windows/Fonts/malgun.ttf").get_name()
    rc('font', family = font_name)
else:
    rc('font',family='AppleGothic')
    
matplotlib.rcParams['axes.unicode_minus']=False
%matplotlib inline

In [4]:
# 데이터프레임은 시트가 여러개이므로, csv 파일로 불러오는 것이 불가능하여, 리스트에 요소별로 데이터프레임을 추가하고, concat하는 방식으로 하였습니다.
df = []
for i in range(9):
    df.append(pd.read_excel('BDAA_1회 내부 공모전 데이터.xlsx', sheet_name=i))

In [5]:
df = pd.concat([df[i] for i in range(9)], axis = 0)

In [6]:
df.head()

주문접수일시              주문번호    유입매체  주문구분      유입채널        회원구분  \
0  2022-01-31 23:59:43  2022013116206973  MO WEB  일반주문  디폴트 채널상세  LPOINT통합회원   
1  2022-01-31 23:44:09  2022013116204996  MO WEB  일반주문  디폴트 채널상세  LPOINT통합회원   
2  2022-01-31 23:44:09  2022013116204996  MO WEB  일반주문  디폴트 채널상세  LPOINT통합회원   
3  2022-01-31 23:35:27  2022013116204259     APP  일반주문  디폴트 채널상세  LPOINT통합회원   
4  2022-01-31 23:21:06  2022013116202890     APP  일반주문  디폴트 채널상세  LPOINT통합회원   

        아이디   전시유입코드              상품주문번호  유형  ...  주문판매유형       상품유형      브랜드  \
0  alls****      NaN  2022013116206973_1  주문  ...  일반판매상품  일반판매_일반상품  씨제이제일제당   
1  okok****      CHT  2022013116204996_1  주문  ...  일반판매상품  일반판매_일반상품  씨제이제일제당   
2  okok****      CHT  2022013116204996_2  주문  ...  일반판매상품  일반판매_일반상품  씨제이제일제당   
3  fire****    SCH스팸  2022013116204259_1  주문  ...  일반판매상품  일반판매_일반상품  씨제이제일제당   
4  dkfm****  SCH비비고국  2022013116202890_1  주문  ...  일반판매상품  일반판매_일반상품  씨제이제일제당   

     판매가 수량                    표준카테고리 모델번호 롯데맴버스오프라인고객번호 롯데맴버스온라인고객번호 회원번호  
0  24900  1   냉장/냉동/간편식 > 떡볶이/만두 > 만두  NaN           NaN          NaN  NaN  
1  26050  1      가공식품 > 장류 > 고추장/초고추장  NaN           NaN          NaN  NaN  
2  22620  1   냉장/냉동/간편식 > 떡볶이/만두 > 만두  NaN           NaN          NaN  NaN  
3  40000  1       가공식품 > 통조림/캔 > 햄통조림  NaN           NaN          NaN  NaN  
4  22030  1  냉장/냉동/간편식 > 국/탕/찌개 > 국/탕  NaN           NaN          NaN  NaN  

[5 rows x 41 columns]

In [7]:
df = df.sort_values('주문접수일시')
df

주문접수일시              주문번호    유입매체  주문구분  \
3266  2022-01-01 00:17:11  2022010111864975  MO WEB  일반주문   
3265  2022-01-01 00:38:13  2022010111866837  MO WEB  일반주문   
3264  2022-01-01 00:38:42  2022010111866938     APP  일반주문   
3263  2022-01-01 02:54:53  2022010111873735     APP  일반주문   
3262  2022-01-01 02:56:35  2022010111873774     APP  일반주문   
...                   ...               ...     ...   ...   
4     2022-09-30 23:52:25  2022093018638708  MO WEB  일반주문   
3     2022-09-30 23:52:41  2022093018638709     APP  일반주문   
2     2022-09-30 23:56:28  2022093018639349     APP  일반주문   
1     2022-09-30 23:58:52  2022093018639892     APP  일반주문   
0     2022-09-30 23:59:27  2022093018640098     APP  일반주문   

                       유입채널        회원구분           아이디     전시유입코드  \
3266               디폴트 채널상세  LPOINT통합회원      jsda****  SCH비비고갈비탕   
3265  [다음카카오] PCS 가격비교_롯데ON  LPOINT통합회원      kim6****        CHT   
3264               디폴트 채널상세  LPOINT통합회원      ggg9****  SCH쁘띠첼복숭아   
3263               디폴트 채널상세  LPOINT통합회원      savi****        RSI   
3262               디폴트 채널상세  LPOINT통합회원      savi****        RSI   
...                     ...         ...           ...        ...   
4       [네이버] PCS 가격비교_롯데ON  LPOINT통합회원       cha2844        CHT   
3                  디폴트 채널상세  LPOINT통합회원        wocns2        CHT   
2                  디폴트 채널상세  LPOINT통합회원       xraydos        RSI   
1          [링크프라이스] CPS_롯데온  LPOINT통합회원       badamit        NaN   
0                  디폴트 채널상세  LPOINT통합회원  simsimfreeya    SCH사골곰탕   

                  상품주문번호        유형  ...  주문판매유형       상품유형      브랜드    판매가 수량  \
3266  2022010111864975_1  취소(주문취소)  ...  일반판매상품  일반판매_일반상품  씨제이제일제당  18830  1   
3265  2022010111866837_1        주문  ...  일반판매상품  일반판매_일반상품  씨제이제일제당  16240  1   
3264  2022010111866938_5  취소(주문취소)  ...  일반판매상품  일반판매_일반상품  씨제이제일제당  22300  1   
3263  2022010111873735_1        주문  ...  일반판매상품  일반판매_일반상품  씨제이제일제당  33240  1   
3262  2022010111873774_1        주문  ...  일반판매상품  일반판매_일반상품  씨제이제일제당  33240  1   
...                  ...       ...  ...     ...        ...      ...    ... ..   
4     2022093018638708_1        주문  ...  일반판매상품  일반판매_일반상품   CJ제일제당  17400  1   
3     2022093018638709_1        주문  ...  일반판매상품  일반판매_일반상품   CJ제일제당  24500  2   
2     2022093018639349_1        주문  ...  일반판매상품  일반판매_일반상품   CJ제일제당  37000  1   
1     2022093018639892_1        주문  ...  일반판매상품  일반판매_일반상품   CJ제일제당  38000  1   
0     2022093018640098_1        주문  ...  일반판매상품  일반판매_일반상품   CJ제일제당  24500  2   

                          표준카테고리 모델번호 롯데맴버스오프라인고객번호 롯데맴버스온라인고객번호          회원번호  
3266    냉장/냉동/간편식 > 국/탕/찌개 > 국/탕  NaN           NaN          NaN           NaN  
3265       가공식품 > 장류 > 된장/쌈장/청국장  NaN           NaN          NaN           NaN  
3264        과자/간식 > 디저트/케이크 > 푸딩  NaN           NaN          NaN           NaN  
3263  냉장/냉동/간편식 > 즉석밥/죽/스프 > 즉석밥  NaN           NaN          NaN           NaN  
3262  냉장/냉동/간편식 > 즉석밥/죽/스프 > 즉석밥  NaN           NaN          NaN           NaN  
...                          ...  ...           ...          ...           ...  
4         가공식품 > 통조림/캔 > 닭가슴살통조림  NaN           NaN          NaN  1.000739e+10  
3       냉장/냉동/간편식 > 국/탕/찌개 > 국/탕  NaN           NaN          NaN  1.001256e+10  
2       냉장/냉동/간편식 > 국/탕/찌개 > 국/탕  NaN           NaN          NaN  1.001334e+10  
1       냉장/냉동/간편식 > 국/탕/찌개 > 국/탕  NaN           NaN          NaN  1.001083e+10  
0       냉장/냉동/간편식 > 국/탕/찌개 > 국/탕  NaN           NaN          NaN  1.000204e+10  

[102077 rows x 41 columns]

In [8]:
df = df.drop(['진행상태 (약식)', '모델번호', '롯데맴버스오프라인고객번호', '롯데맴버스온라인고객번호', '회원번호'], axis = 1)

- 칼럼이 너무 많아서, 단일 값이거나, 널값인 칼럼은 제거하도록 하겠습니다.

In [9]:
for i in range(len(df.columns)):
    print(df.columns[i], len(df[df.columns[i]].unique()))

주문접수일시 92771
주문번호 95036
유입매체 5
주문구분 2
유입채널 551
회원구분 5
아이디 53401
전시유입코드 3389
상품주문번호 102076
유형 4
배송회수구분 2
진행단계 6
구매확정 3
주문완료일시 92468
구매확정일시 38160
출고지시일 278
예외처리구분 4
상태처리일시 284
배송상품유형 1
배송처리유형 1
배송수단 1
배송예약구분 1
배송회차 1
전시상품명 2188
전시단품명 489
판매자상품번호 1662
판매자상품명 2185
판매자단품번호 2096
판매자단품명 489
추가옵션 1
주문판매유형 1
상품유형 1
브랜드 3
판매가 1651
수량 37
표준카테고리 65


In [10]:
df = df.drop(['배송수단', '배송상품유형', '배송처리유형', '배송예약구분', '배송회차', '추가옵션', '주문판매유형', '상품유형', '브랜드'], axis=1)

In [11]:
df['구매확정'].unique()

array(['구매미확정', '구매확정', '구매확정연장'], dtype=object)

In [12]:
df['예외처리구분'].unique()

array([nan, '구매확정후취소', '미수령신고접수철회요청', '발송약정'], dtype=object)

In [13]:
# 예외처리구분이 발송약정인 데이터 1개는 구매미확정, 교환 유형에 해당되는군요! -> 구매 미확정 항목에 넣어주겠습니다!

df[df['예외처리구분'] == '발송약정']

주문접수일시              주문번호 유입매체  주문구분      유입채널        회원구분  \
1044  2022-09-27 11:14:18  2022092718188481  APP  일반주문  디폴트 채널상세  LPOINT통합회원   

          아이디 전시유입코드              상품주문번호  유형  ...      상태처리일시  \
1044  sericup    MLT  2022092718188481_1  교환  ...  2022-10-06   

                           전시상품명 전시단품명       판매자상품번호  \
1044  해찬들 우리쌀로만든태양초매운고추장2kg X 2개  단일상품  LO1880319013   

                          판매자상품명                  판매자단품번호 판매자단품명    판매가 수량  \
1044  해찬들 우리쌀로만든태양초매운고추장2kg X 2개  LO1880319013_1880319014   단일상품  25330  1   

                    표준카테고리  
1044  가공식품 > 장류 > 고추장/초고추장  

[1 rows x 27 columns]

In [14]:
df.loc[df['예외처리구분'].notna()]

주문접수일시              주문번호    유입매체  주문구분  \
3137   2022-01-03 14:35:32  2022010312252348     APP  일반주문   
2884   2022-01-06 17:51:38  2022010612798197     APP  일반주문   
2749   2022-01-07 08:02:07  2022010712868682  MO WEB  일반주문   
2398   2022-01-09 00:32:21  2022010913107228     APP  일반주문   
96     2022-01-26 17:52:01  2022012615687122  MO WEB  선물하기   
1489   2022-02-07 17:07:18  2022020717164453     APP  일반주문   
19     2022-04-30 21:49:23  2022043019140237      PC  일반주문   
10611  2022-05-08 12:53:54  2022050810196157  MO WEB  일반주문   
4477   2022-05-25 13:30:17  2022052512522342  MO WEB  일반주문   
4390   2022-05-25 13:50:04  2022052512525047  MO WEB  일반주문   
15311  2022-07-01 12:58:53  2022070117367209     APP  일반주문   
8688   2022-07-21 11:21:55  2022072119957104     APP  일반주문   
26090  2022-08-03 23:55:57  2022080311625906     APP  일반주문   
26091  2022-08-03 23:55:57  2022080311625906     APP  일반주문   
23430  2022-08-05 08:56:33  2022080511789001     APP  일반주문   
18037  2022-08-06 23:38:03  2022080612009109     APP  일반주문   
17506  2022-08-07 00:07:23  2022080712012990  MO WEB  일반주문   
13663  2022-08-07 11:12:12  2022080712049566     APP  일반주문   
2201   2022-08-25 10:00:40  2022082514097676      PC  일반주문   
1325   2022-08-29 23:57:14  2022082914680634      PC  일반주문   
438    2022-08-31 16:01:21  2022083114940348     APP  일반주문   
136    2022-08-31 22:11:26  2022083114991155     APP  일반주문   
5463   2022-09-05 02:23:30  2022090515569855  MO WEB  일반주문   
1044   2022-09-27 11:14:18  2022092718188481     APP  일반주문   

                        유입채널        회원구분         아이디  \
3137                디폴트 채널상세  LPOINT통합회원    hyun****   
2884     [네이버] PCS 가격비교_롯데ON  LPOINT통합회원    muct****   
2749     [에누리] PCS 가격비교_롯데ON  LPOINT통합회원    ldk0****   
2398                디폴트 채널상세  LPOINT통합회원    soki****   
96                  디폴트 채널상세  LPOINT통합회원    mere****   
1489                디폴트 채널상세  LPOINT통합회원    wlsr****   
19       [네이버] PCS 가격비교_롯데ON  LPOINT통합회원    powe****   
10611    [네이버] PCS 가격비교_롯데ON  LPOINT통합회원    sam1****   
4477                 유실 채널상세  LPOINT통합회원    odam****   
4390                 유실 채널상세  LPOINT통합회원    odam****   
15311               디폴트 채널상세  LPOINT통합회원    Kong****   
8688                디폴트 채널상세  LPOINT통합회원    lthoon88   
26090               디폴트 채널상세  LPOINT통합회원    jinu7979   
26091               디폴트 채널상세  LPOINT통합회원    jinu7979   
23430               디폴트 채널상세  LPOINT통합회원      naxe78   
18037               디폴트 채널상세  LPOINT통합회원  dlgpal7520   
17506               디폴트 채널상세  LPOINT통합회원  mastergoda   
13663               디폴트 채널상세  LPOINT통합회원   pmistakeq   
2201     [에누리] PCS 가격비교_롯데ON  LPOINT통합회원  urimiracle   
1325   [다음카카오] PCS 가격비교_롯데ON      소셜간편회원         NaN   
438                  유실 채널상세  LPOINT통합회원    hyosik79   
136                 디폴트 채널상세  LPOINT통합회원      q1azzx   
5463     [네이버] PCS 가격비교_롯데ON  LPOINT통합회원     zoon119   
1044                디폴트 채널상세  LPOINT통합회원     sericup   

                                                  전시유입코드              상품주문번호  \
3137                                                 SCT  2022010312252348_2   
2884   CHT&amp;NaPm=ct=ky2q6h8w|ci=7584e3fed016cd636e...  2022010612798197_1   
2749                                                 CHT  2022010712868682_1   
2398                                            MAT29675  2022010913107228_2   
96                                                   RSI  2022012615687122_1   
1489                                                 MLT  2022020717164453_1   
19                                                   CHT  2022043019140237_1   
10611                                                MLT  2022050810196157_1   
4477                                                 CHT  2022052512522342_1   
4390                                                 CHT  2022052512525047_1   
15311                                           SCH비비고만두  2022070117367209_3   
8688                                            MAT12905  2022072119957104_1   
26090                       

- 구매 미확정 상품 중 취소된 상품, 
- 구매 확정인데 반품된 상품, 
- '구매확정' 및 '구매확정연장' 상품 중 예외처리구분에 값이 있는 상품 (미수령신고접수철회요청, 구매확정후취소)
df_canceled 데이터프레임에 따로 저장할게요~!

In [15]:
df['유형'].unique()

array(['취소(주문취소)', '주문', '반품', '교환'], dtype=object)

In [16]:
df_canceled = df.loc[(df['구매확정'] == '구매미확정') | (df['유형'] == '반품') | (df['예외처리구분'].notna())]

In [17]:
# 유형에 "주문"은 뭘까요?? 취소된 데이터일까요~~?

df_canceled['유형'].unique()

array(['취소(주문취소)', '반품', '주문', '교환'], dtype=object)

유형이 주문, 예외처리구분에 값이 없는 주문 3건은, 주문접수가 9월이며,  취소 여부가 불확실한 데이터군요! -> 취소한 제품들로 성급하게 판단하면 안될 거 같아요. 얘네는 df_canceled에서 빼야겠어요!

In [18]:
df_canceled.loc[(df_canceled['유형'] == '주문') & (df_canceled['예외처리구분'].isna())]

주문접수일시              주문번호    유입매체  주문구분              유입채널  \
3739  2022-09-18 20:45:03  2022091817078734     APP  일반주문          디폴트 채널상세   
283   2022-09-30 15:33:30  2022093018590221  MO WEB  일반주문  [링크프라이스] CPS_롯데온   
182   2022-09-30 18:54:37  2022093018607386     APP  일반주문  [링크프라이스] CPS_롯데온   

            회원구분         아이디   전시유입코드              상품주문번호  유형  ...  \
3739  LPOINT통합회원      sirube  SCH햇반잡곡  2022091817078734_1  주문  ...   
283       소셜간편회원         NaN  SCH사골곰탕  2022093018590221_1  주문  ...   
182   LPOINT통합회원  newbabypig      NaN  2022093018607386_1  주문  ...   

          상태처리일시                          전시상품명               전시단품명  \
3739  2022-10-15             햇반 매일찰잡곡밥 210G x24                단일상품   
283   2022-10-13             비비고 사골곰탕 500g x18봉                단일상품   
182   2022-10-04  [L포인트 3000p 증정]LAST DAY EVENT  비비고 사골곰탕 500g x18봉   

           판매자상품번호                           판매자상품명                  판매자단품번호  \
3739  LO1135620329               햇반 매일찰잡곡밥 210G x24  LO1135620329_1135620330   
283   LO1032764663               비비고 사골곰탕 500g x18봉  LO1032764663_1032764664   
182   LO1978869746  [L포인트 3000p 증정]★LAST DAY EVENT★  LO1978869746_1978869755   

                  판매자단품명    판매가 수량                      표준카테고리  
3739                단일상품  43060  1  냉장/냉동/간편식 > 즉석밥/죽/스프 > 즉석밥  
283                 단일상품  24600  1    냉장/냉동/간편식 > 국/탕/찌개 > 국/탕  
182   비비고 사골곰탕 500g x18봉  24500  1    냉장/냉동/간편식 > 국/탕/찌개 > 국/탕  

[3 rows x 27 columns]

- 아까 모았던 구매 미확정 데이터 중 유형이 '취소(주문취소)', '반품' 이거나, 유형은 '주문'인데 예외처리구분이 ('미수령신고접수철회요청', '구매확정후취소')인 데이터를 df_canceled에 모아주겠습니다.

In [19]:
df_canceled = df_canceled.loc[(df_canceled['유형'] == '취소(주문취소)') | ((df_canceled['유형'] == '반품')) | ((df_canceled['예외처리구분'].notna()))]

- 전체 데이터 102077개중 주문 취소 및 구매 확정후 취소, 미수령신고접수 후 철회요청 된 데이터 8702개가 df_canceled에 저장되었군요!


In [20]:
df_canceled

주문접수일시              주문번호    유입매체  주문구분              유입채널  \
3266  2022-01-01 00:17:11  2022010111864975  MO WEB  일반주문          디폴트 채널상세   
3264  2022-01-01 00:38:42  2022010111866938     APP  일반주문          디폴트 채널상세   
3249  2022-01-01 17:16:48  2022010111935204     APP  일반주문          디폴트 채널상세   
3228  2022-01-02 13:03:20  2022010212023381  MO WEB  일반주문          디폴트 채널상세   
3227  2022-01-02 13:03:20  2022010212023381  MO WEB  일반주문          디폴트 채널상세   
...                   ...               ...     ...   ...               ...   
45    2022-09-30 22:49:16  2022093018630370     APP  일반주문  [링크프라이스] CPS_롯데온   
38    2022-09-30 22:56:31  2022093018631261     APP  일반주문          디폴트 채널상세   
32    2022-09-30 23:02:38  2022093018631923     APP  일반주문  [링크프라이스] CPS_롯데온   
30    2022-09-30 23:05:50  2022093018632420     APP  일반주문          디폴트 채널상세   
6     2022-09-30 23:49:32  2022093018638203     APP  일반주문  [링크프라이스] CPS_롯데온   

            회원구분       아이디     전시유입코드              상품주문번호        유형  ...  \
3266  LPOINT통합회원  jsda****  SCH비비고갈비탕  2022010111864975_1  취소(주문취소)  ...   
3264  LPOINT통합회원  ggg9****  SCH쁘띠첼복숭아  2022010111866938_5  취소(주문취소)  ...   
3249  LPOINT통합회원  love****     SCH파래김  2022010111935204_1  취소(주문취소)  ...   
3228  LPOINT통합회원  pilj****   BSGP2566  2022010212023381_2  취소(주문취소)  ...   
3227  LPOINT통합회원  pilj****   BSGP2566  2022010212023381_1  취소(주문취소)  ...   
...          ...       ...        ...                 ...       ...  ...   
45    LPOINT통합회원  pete****        NaN  2022093018630370_1  취소(주문취소)  ...   
38    LPOINT통합회원  yeop****        NaN  2022093018631261_1  취소(주문취소)  ...   
32    LPOINT통합회원  tae0****        CHT  2022093018631923_1  취소(주문취소)  ...   
30    LPOINT통합회원  dmsd****        RSI  2022093018632420_1  취소(주문취소)  ...   
6     LPOINT통합회원  20re****        NaN  2022093018638203_1  취소(주문취소)  ...   

          상태처리일시                                전시상품명               전시단품명  \
3266  2022-01-03                   CJ 비비고 갈비탕 400g x4                단일상품   
3264  2022-01-03         [CJ제일제당] 쁘띠첼 복숭아 90g*9번들X2EA                단일상품   
3249  2022-01-03      [CJ직배송]CJ 명가김 파래김 4G*16EA  X 2개                단일상품   
3228  2022-01-02  [특별이벤트] 햇반 BIG 컵반 스팸 2종 X 3개 + 주식선물                단일상품   
3227  2022-01-02     [특별이벤트]햇반 BIG 컵반 마요 2종 X 3개+주식선물                단일상품   
...          ...                                  ...                 ...   
45    2022-09-30        [L포인트 3000p 증정]LAST DAY EVENT  비비고 사골곰탕 500g x18봉   
38    2022-10-04                   비비고 사골곰탕 500g x18봉                단일상품   
32    2022-09-30                   비비고 사골곰탕 500g x18봉                단일상품   
30    2022-10-03                   비비고 사골곰탕 500g x18봉                단일상품   
6     2022-10-04                   비비고 사골곰탕 500g x18봉                단일상품   

           판매자상품번호                               판매자상품명  \
3266  LO1031678373                   CJ 비비고 갈비탕 400g x4   
3264  LO1354545470         [CJ제일제당] 쁘띠첼 복숭아 90g*9번들X2EA   
3249  LO1511816458      [CJ직배송]CJ 명가김 파래김 4G*16EA  X 2개   
3228  LO1631718294  [특별이벤트] 햇반 BIG 컵반 스팸 2종 X 3개 + 주식선물   
3227  LO1630046529     [특별이벤트]햇반 BIG 컵반 마요 2종 X 3개+주식선물   
...            ...                                  ...   
45    LO1978869746      [L포인트 3000p 증정]★LAST DAY EVENT★   
38    LO1032764663                   비비고 사골곰탕 500g x18봉   
32    LO1032764663                   비비고 사골곰탕 500g x18봉   
30    LO1032764663                   비비고 사골곰탕 500g x18봉   
6     LO1032764663                   비비고 사골곰탕 500g x18봉   

                      판매자단품번호              판매자단품명    판매가 수량  \
3266  LO1031678373_1031678374                단일상품  18830  1   
3264  LO1354545470_1354545471                단일상품  22300  1   
3249  LO1511816458_1511816459                단일상품   6960  3   
3228  LO1631718294_1631718295                단일상품  19920  1   
3227  LO1630046529_1630046530                단일상품  19920  1   
...                       ...                 ...    ... ..   
45    LO1978869746_1978869755  비비고 사골곰탕 500g x18봉  24500  2   
38    LO1032764663_

In [21]:
df_canceled['판매액'] = df_canceled['판매가'] * df_canceled['수량']

In [22]:
df_canceled['주문접수일시'] = pd.to_datetime(df_canceled['주문접수일시'], format = "%Y-%m-%d %H:%M:%S")

In [23]:
df_canceled['주문월'] = df_canceled['주문접수일시'].dt.month

In [24]:
df_canceled = df_canceled.reset_index()

In [25]:
df_canceled = df_canceled.drop(['index'], axis=1)

In [26]:
df_canceled

주문접수일시              주문번호    유입매체  주문구분              유입채널  \
0    2022-01-01 00:17:11  2022010111864975  MO WEB  일반주문          디폴트 채널상세   
1    2022-01-01 00:38:42  2022010111866938     APP  일반주문          디폴트 채널상세   
2    2022-01-01 17:16:48  2022010111935204     APP  일반주문          디폴트 채널상세   
3    2022-01-02 13:03:20  2022010212023381  MO WEB  일반주문          디폴트 채널상세   
4    2022-01-02 13:03:20  2022010212023381  MO WEB  일반주문          디폴트 채널상세   
...                  ...               ...     ...   ...               ...   
8697 2022-09-30 22:49:16  2022093018630370     APP  일반주문  [링크프라이스] CPS_롯데온   
8698 2022-09-30 22:56:31  2022093018631261     APP  일반주문          디폴트 채널상세   
8699 2022-09-30 23:02:38  2022093018631923     APP  일반주문  [링크프라이스] CPS_롯데온   
8700 2022-09-30 23:05:50  2022093018632420     APP  일반주문          디폴트 채널상세   
8701 2022-09-30 23:49:32  2022093018638203     APP  일반주문  [링크프라이스] CPS_롯데온   

            회원구분       아이디     전시유입코드              상품주문번호        유형  ...  \
0     LPOINT통합회원  jsda****  SCH비비고갈비탕  2022010111864975_1  취소(주문취소)  ...   
1     LPOINT통합회원  ggg9****  SCH쁘띠첼복숭아  2022010111866938_5  취소(주문취소)  ...   
2     LPOINT통합회원  love****     SCH파래김  2022010111935204_1  취소(주문취소)  ...   
3     LPOINT통합회원  pilj****   BSGP2566  2022010212023381_2  취소(주문취소)  ...   
4     LPOINT통합회원  pilj****   BSGP2566  2022010212023381_1  취소(주문취소)  ...   
...          ...       ...        ...                 ...       ...  ...   
8697  LPOINT통합회원  pete****        NaN  2022093018630370_1  취소(주문취소)  ...   
8698  LPOINT통합회원  yeop****        NaN  2022093018631261_1  취소(주문취소)  ...   
8699  LPOINT통합회원  tae0****        CHT  2022093018631923_1  취소(주문취소)  ...   
8700  LPOINT통합회원  dmsd****        RSI  2022093018632420_1  취소(주문취소)  ...   
8701  LPOINT통합회원  20re****        NaN  2022093018638203_1  취소(주문취소)  ...   

                   전시단품명       판매자상품번호                               판매자상품명  \
0                   단일상품  LO1031678373                   CJ 비비고 갈비탕 400g x4   
1                   단일상품  LO1354545470         [CJ제일제당] 쁘띠첼 복숭아 90g*9번들X2EA   
2                   단일상품  LO1511816458      [CJ직배송]CJ 명가김 파래김 4G*16EA  X 2개   
3                   단일상품  LO1631718294  [특별이벤트] 햇반 BIG 컵반 스팸 2종 X 3개 + 주식선물   
4                   단일상품  LO1630046529     [특별이벤트]햇반 BIG 컵반 마요 2종 X 3개+주식선물   
...                  ...           ...                                  ...   
8697  비비고 사골곰탕 500g x18봉  LO1978869746      [L포인트 3000p 증정]★LAST DAY EVENT★   
8698                단일상품  LO1032764663                   비비고 사골곰탕 500g x18봉   
8699                단일상품  LO1032764663                   비비고 사골곰탕 500g x18봉   
8700                단일상품  LO1032764663                   비비고 사골곰탕 500g x18봉   
8701                단일상품  LO1032764663                   비비고 사골곰탕 500g x18봉   

                      판매자단품번호              판매자단품명    판매가 수량  \
0     LO1031678373_1031678374                단일상품  18830  1   
1     LO1354545470_1354545471                단일상품  22300  1   
2     LO1511816458_1511816459                단일상품   6960  3   
3     LO1631718294_1631718295                단일상품  19920  1   
4     LO1630046529_1630046530                단일상품  19920  1   
...                       ...                 ...    ... ..   
8697  LO1978869746_1978869755  비비고 사골곰탕 500g x18봉  24500  2   
8698  LO1032764663_1032764664                단일상품  24600  2   
8699  LO1032764663_1032764664                단일상품  24600  1   
8700  LO1032764663_1032764664                단일상품  24600  1   
8701  LO1032764663_1032764664                단일상품  24600  2   

                          표준카테고리    판매액 주문월  
0       냉장/냉동/간편식 > 국/탕/찌개 > 국/탕  18830   1  
1           과자/간식 > 디저트/케이크 > 푸딩  22300   1  
2      수산/건어물 > 김/해조류 > 조미김/도시락김  20880   1  
3     냉장/냉동/간편식 > 즉석밥/죽/스프 > 즉석밥  19920   1  
4     냉장/냉동/간편식 > 즉석밥/죽/스프 > 즉석밥  19920   1  
...                          ...    ...  ..  
8697    냉장/냉동/간편식 > 국/탕/찌개 > 국/탕  49000   9  
8698    냉장/냉동/간편식 > 국/탕/찌개 > 국/탕  49200   9  
8699    냉장/냉동/간편식 > 국/탕/찌개 > 국/탕  24600   9  
87

In [28]:
# df_canceled.to_csv('구매미확정 및 취소된 제품.csv')

In [29]:
df = pd.read_csv('구매미확정 및 취소된 제품.csv')

In [30]:
df

주문접수일시              주문번호    유입매체  주문구분              유입채널  \
0     2022-01-01 00:17:11  2022010111864970  MO WEB  일반주문          디폴트 채널상세   
1     2022-01-01 00:38:42  2022010111866940     APP  일반주문          디폴트 채널상세   
2     2022-01-01 17:16:48  2022010111935200     APP  일반주문          디폴트 채널상세   
3     2022-01-02 13:03:20  2022010212023380  MO WEB  일반주문          디폴트 채널상세   
4     2022-01-02 13:03:20  2022010212023380  MO WEB  일반주문          디폴트 채널상세   
...                   ...               ...     ...   ...               ...   
8697  2022-09-30 22:49:16  2022093018630370     APP  일반주문  [링크프라이스] CPS_롯데온   
8698  2022-09-30 22:56:31  2022093018631260     APP  일반주문          디폴트 채널상세   
8699  2022-09-30 23:02:38  2022093018631920     APP  일반주문  [링크프라이스] CPS_롯데온   
8700  2022-09-30 23:05:50  2022093018632420     APP  일반주문          디폴트 채널상세   
8701  2022-09-30 23:49:32  2022093018638200     APP  일반주문  [링크프라이스] CPS_롯데온   

            회원구분       아이디     전시유입코드              상품주문번호        유형  ...  \
0     LPOINT통합회원  jsda****  SCH비비고갈비탕  2022010111864975_1  취소(주문취소)  ...   
1     LPOINT통합회원  ggg9****  SCH쁘띠첼복숭아  2022010111866938_5  취소(주문취소)  ...   
2     LPOINT통합회원  love****     SCH파래김  2022010111935204_1  취소(주문취소)  ...   
3     LPOINT통합회원  pilj****   BSGP2566  2022010212023381_2  취소(주문취소)  ...   
4     LPOINT통합회원  pilj****   BSGP2566  2022010212023381_1  취소(주문취소)  ...   
...          ...       ...        ...                 ...       ...  ...   
8697  LPOINT통합회원  pete****        NaN  2022093018630370_1  취소(주문취소)  ...   
8698  LPOINT통합회원  yeop****        NaN  2022093018631261_1  취소(주문취소)  ...   
8699  LPOINT통합회원  tae0****        CHT  2022093018631923_1  취소(주문취소)  ...   
8700  LPOINT통합회원  dmsd****        RSI  2022093018632420_1  취소(주문취소)  ...   
8701  LPOINT통합회원  20re****        NaN  2022093018638203_1  취소(주문취소)  ...   

                   전시단품명       판매자상품번호                               판매자상품명  \
0                   단일상품  LO1031678373                   CJ 비비고 갈비탕 400g x4   
1                   단일상품  LO1354545470         [CJ제일제당] 쁘띠첼 복숭아 90g*9번들X2EA   
2                   단일상품  LO1511816458      [CJ직배송]CJ 명가김 파래김 4G*16EA  X 2개   
3                   단일상품  LO1631718294  [특별이벤트] 햇반 BIG 컵반 스팸 2종 X 3개 + 주식선물   
4                   단일상품  LO1630046529     [특별이벤트]햇반 BIG 컵반 마요 2종 X 3개+주식선물   
...                  ...           ...                                  ...   
8697  비비고 사골곰탕 500g x18봉  LO1978869746      [L포인트 3000p 증정]★LAST DAY EVENT★   
8698                단일상품  LO1032764663                   비비고 사골곰탕 500g x18봉   
8699                단일상품  LO1032764663                   비비고 사골곰탕 500g x18봉   
8700                단일상품  LO1032764663                   비비고 사골곰탕 500g x18봉   
8701                단일상품  LO1032764663                   비비고 사골곰탕 500g x18봉   

                      판매자단품번호              판매자단품명    판매가 수량  \
0     LO1031678373_1031678374                단일상품  18830  1   
1     LO1354545470_1354545471                단일상품  22300  1   
2     LO1511816458_1511816459                단일상품   6960  3   
3     LO1631718294_1631718295                단일상품  19920  1   
4     LO1630046529_1630046530                단일상품  19920  1   
...                       ...                 ...    ... ..   
8697  LO1978869746_1978869755  비비고 사골곰탕 500g x18봉  24500  2   
8698  LO1032764663_1032764664                단일상품  24600  2   
8699  LO1032764663_1032764664                단일상품  24600  1   
8700  LO1032764663_1032764664                단일상품  24600  1   
8701  LO1032764663_1032764664                단일상품  24600  2   

                          표준카테고리    판매액 주문월  
0       냉장/냉동/간편식 > 국/탕/찌개 > 국/탕  18830   1  
1           과자/간식 > 디저트/케이크 > 푸딩  22300   1  
2      수산/건어물 > 김/해조류 > 조미김/도시락김  20880   1  
3     냉장/냉동/간편식 > 즉석밥/죽/스프 > 즉석밥  19920   1  
4     냉장/냉동/간편식 > 즉석밥/죽/스프 > 즉석밥  19920   1  
...                          ...    ...  ..  
8697    냉장/냉동/간편식 > 국/탕/찌개 > 국/탕  49000   9  
8698    냉장/냉동/간편식 > 국/탕/찌개 > 국/탕  49200   9  
8699    냉장/냉동/간편식 > 국/탕/찌개 > 국/탕  246